### PR TITLE
ddl: check if the dropping index is primary key when doing "drop index `primary` on...".

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
Mainly solve the following two problems：

- Case 1
```
    alter-primary-key = true
    create table t (a int, b varchar(10), primary key(a))"
    alter-primary-key = false
    drop index `primary` on t
```
Before this PR:
```
    Query OK, 0 rows affected (0.02 sec)
```
After this PR:
```
    [ddl:8200]Unsupported drop primary key when alter-primary-key is false"
```

- Case 2

```
    alter-primary-key = false
    create table t (a int, b varchar(10), primary key(a))
    drop index `primary` on t
```
Before this PR:
```
    [ddl:1091]index primary doesn't exist
```
After this PR:
```
    [ddl:8200]Unsupported drop primary key when alter-primary-key is false
```

### What is changed and how it works?
Check if the dropping index is the primary key when doing "drop index `primary` on...". 

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

Release note

 - Check if the dropping index is the primary key when doing "drop index `primary` on...".

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -3587,6 +3583,10 @@ func (d *ddl) dropIndex(ctx sessionctx.Context, ti ast.Ident, isPK bool, indexNa
 		return errors.Trace(infoschema.ErrTableNotExists.GenWithStackByArgs(ti.Schema, ti.Name))
 	}
 
+	var isPK bool
+	if indexName.L == strings.ToLower(mysql.PrimaryKeyName) {
```
Comment (wjhuang2016) [inline ddl/ddl_api.go]: We can create a normal index with name `primary`:
```
create table tt(`primary` int);
alter table tt add index (`primary`);
```

Comment (zimulala) [reply ddl/ddl_api.go]: This seems to be the problem that the index name we used to create the index for the 
`primary` column is not compatible with MySQL
```
mysql> create table tt(`primary` int);
Query OK, 0 rows affected (0.02 sec)
mysql> alter table tt add index (`primary`);
Query OK, 0 rows affected (0.03 sec)
Records: 0  Duplicates: 0  Warnings: 0
mysql> show create table tt;
+-------+--------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                             |
+-------+--------------------------------------------------------------------------------------------------------------------------+
| tt    | CREATE TABLE `tt` (
  `primary` int(11) DEFAULT NULL,
  KEY `primary_2` (`primary`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 |
+-------+--------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

```
tidb>  create table tt(`primary` int);
Query OK, 0 rows affected (0.01 sec)
tidb> alter table tt add index (`primary`);
Query OK, 0 rows affected (0.04 sec)
tidb> show create table tt;
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                  |
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------+
| tt    | CREATE TABLE `tt` (
  `primary` int(11) DEFAULT NULL,
  KEY `primary` (`primary`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)
```

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -3588,6 +3584,11 @@ func (d *ddl) dropIndex(ctx sessionctx.Context, ti ast.Ident, isPK bool, indexNa
 	}
 
 	indexInfo := t.Meta().FindIndexByName(indexName.L)
+	var isPK bool
+	if indexName.L == strings.ToLower(mysql.PrimaryKeyName) &&
```
Comment (wjhuang2016) [inline ddl/ddl_api.go]: Suppose that:
```
create table tt(`primary` int, b int);
alter table tt add index (`primary`);
```
What if 
`alter table tt add primary key(b);`?

File: ddl/ddl_api.go [hunk 3]
```diff
@@ -3588,6 +3584,11 @@ func (d *ddl) dropIndex(ctx sessionctx.Context, ti ast.Ident, isPK bool, indexNa
...
+		(indexInfo == nil || indexInfo.Primary) {
+		isPK = true
+	}
```
Comment (Deardrops) [inline ddl/ddl_api.go]: wrap this check logic to function `IsPK()`

File: ddl/ddl_api.go (hunk 2 above)
Comment (zimulala) [reply ddl/ddl_api.go]: I think this question and your last question are a problem. I file the issue of #14243. I think we can handle it in the next PR.

Comment (wjhuang2016) [reply ddl/ddl_api.go]: If we need to follow MySQL name convention, that `primay` is always used by PRIMARY KEY, then we don't need to check `(indexInfo == nil || indexInfo.Primary)`.

Comment (zimulala) [reply ddl/ddl_api.go]: But this problem may occur with indexes that already exist in the old TiDB cluster. I will add some comments here.

File: ddl/ddl_api.go (hunk 3 above)
Comment (zimulala) [reply ddl/ddl_api.go]: This logic is only used here, so I don't think it is necessary to add a new method.

Comment (wjhuang2016) [reply ddl/ddl_api.go]: `indexName.L == strings.ToLower(mysql.PrimaryKeyName)` is enough?

Comment (zimulala) [reply ddl/ddl_api.go]: But this problem may occur with indexes that already exist in the old TiDB cluster. I will add some comments here.

**Pull request merged**